pkg/config/cfgvars.go

```diff
@@ -131,13 +132,29 @@ func NewCfgVars(cobraCmd command, dirs ...string) (*CfgVars, error) {
 		return nil, err
 	}
 
+	euid := os.Geteuid()
+
 	var runDir string
-	if os.Geteuid() == 0 {
+	if euid == 0 {
 		runDir = "/run/k0s"
 	} else {
 		runDir = filepath.Join(dataDir, "run")
 	}
 
+	var statusSocketPath string
+	if cobraCmd != nil {
+		if val, err := cobraCmd.Flags().GetString("status-socket"); err == nil {
+			statusSocketPath = val
+		}
+	}
+	if statusSocketPath == "" {
+		if runtime.GOOS == "windows" || euid == 0 {
+			statusSocketPath = constant.StatusSocketPathDefault
+		} else {
+			statusSocketPath = filepath.Join(runDir, "status.sock")
+		}
```

Comment:
The non-windows fallback is right for every user, as the correct runDir is set based on the euid already.

```suggestion
		if runtime.GOOS == "windows" {
			statusSocketPath = constant.StatusSocketPathDefault
		} else {
			statusSocketPath = filepath.Join(runDir, "status.sock")
		}
```